### DISCUSSION TITLE
Submodule clone/init rate limit?

### DISCUSSION THREAD

**Discussion by amcnicoll** [community/community#23072] [New to GitHub]
<p>We (unfortunately) have a repository with a few dozen submodules.</p>
<p>When cloning this repository fresh, we typically want to init every single submodule. We are finding that the clone stalls out every 4-5 submodules and becomes unresponsive.</p>
<p>Terminating the command and re-starting it does not resolve the issue… but waiting a few minutes <em>does</em>. It then successfully inits another 4-5 submodules, and then stalls again.</p>
<p>This is true whether performing the <code>git submodule init</code> in a command line or in GitKraken (which most team members use).</p>
<p>Is GitHub imposing some kind of rate limit here? Any other explanations? We have many users and machines that occasionally need a fresh cloning, and this makes it very tedious to get a fresh copy of the repo with all submodules in place.</p>

**Comment by queenofcorgis** [comment]
<p>Many things could cause this. Mostly, I think, are in the your Git configuration or in the network between yourself and the server. If you’re using Git over SSH and your repository is on <a href="http://github.com/">GitHub.com</a>, they can run this to isolate a little:</p>
<p><code>for i in {1..40}; do ssh -T git@github.com; done</code></p>
<p>That should return something like this 40 times:</p>
<p><code>Hi username! You've successfully authenticated, but GitHub does not provide shell access.</code></p>
<p>If that <code>for</code> loop above starts failing after the 20th success, you can isolate the problem further by testing against a host GitHub does not operate:</p>
<p><code>for i in {1..40}; do ssh -T test@5.1.66.10; done</code></p>
<p>That one should return <code>Permission denied (publickey).</code> 40 times.</p>

**Comment by amcnicoll** [comment, marked as answer]
<p>Thank you - this indeed turned out to be an issue with our network. Problem solved!</p>

**Comment by queenofcorgis** [comment]
<p>Glad to hear so! Thanks for the update :stars:</p>
